### PR TITLE
Update karmada's k8s components to 1.25.2

### PR DESCRIPTION
**Pull request by jwcesign**
Signed-off-by: jwcesign <jiangwei115@huawei.com>

**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:
Update karmada's dependent k8s karmada to latest version/1.25.2

Test results:
`local-up-karmada.sh` deploy way:
image
image

`kubectl karmada init` deploy way:
image

`helm` deploy way:
image

**Which issue(s) this PR fixes**:
Update karmada's dependent k8s karmada to latest version/1.25.2

**Special notes for your reviewer**:
none

**Does this PR introduce a user-facing change?**:

```release-note
karmadactl/chart: if karmada is installed by using karmadactl or helm chart with default configuration, the image version of karmada-kube-controller-manager/karmada-apiserver will be kube-controller-manager:v1.25.2/kube-apiserver:v1.25.2.
```

### CODE REVIEW

**Comment by RainbowMango** [comment]
@jwcesign Does this ready for moving forward? there is still a `WIP` in the PR title.

**Comment by jwcesign** [comment]
> @jwcesign Does this ready for moving forward? there is still a `WIP` in the PR title.

Stil a WIP PR, I doing about kubectl

**Comment by XiShanYongYe-Chang** [comment]
This pr needs to wait for #2547 and #2548 
/hold

File: pkg/karmadactl/cmdinit/kubernetes/deploy.go
Comment (RainbowMango) [reply pkg/karmadactl/cmdinit/kubernetes/deploy.go:180]: > This directory is deleted in previous steps, should create it again.

Why? do you mean here is a bug?

File: pkg/karmadactl/cmdinit/kubernetes/deployments.go
Comment (RainbowMango) [inline pkg/karmadactl/cmdinit/kubernetes/deployments.go:561]: Why remove this by this PR?

File: pkg/karmadactl/cmdinit/utils/examples.go
Comment (RainbowMango) [inline pkg/karmadactl/cmdinit/utils/examples.go:187]: nice finding.

**Comment by RainbowMango** [comment]
By the way, are you going to update the version for the chart?
I'm asking because I see this patch including both development and karmadactl, just the chart left. I'm ok to do it by separated PR, just make sure the chart part is still on your plan.

File: pkg/karmadactl/cmdinit/kubernetes/deployments.go
Comment (jwcesign) [reply pkg/karmadactl/cmdinit/kubernetes/deployments.go:561]: > By the way, are you going to update the version for the chart? I'm asking because I see this patch including both development and karmadactl, just the chart left. I'm ok to do it by separated PR, just make sure the chart part is still on your plan.

On the way

File: pkg/karmadactl/cmdinit/kubernetes/deploy.go
Comment (jwcesign) [reply pkg/karmadactl/cmdinit/kubernetes/deploy.go:180]: Yes, it's a bug, fixed in https://github.com/karmada-io/karmada/pull/2548

File: pkg/karmadactl/cmdinit/kubernetes/deployments.go
Comment (jwcesign) [reply pkg/karmadactl/cmdinit/kubernetes/deployments.go:561]: in this PR, will rebase after it merged: https://github.com/karmada-io/karmada/pull/2547

**Comment by RainbowMango** [comment]
I think we need a release note as the behavior changed for both `karmadactl` and `chart` that they now default to a newer version.

**Comment by RainbowMango** [comment]
I just found that Kuberentes has released v1.25.2 which fixed a `kube-apiserver` bug. ([See this notes](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.25.md#changes-by-kind)).

So, could you please update to v1.25.2?

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**
